AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +3814,263 @@ var interpolate = function(fromValue, toValue, progress)
+                                                                              modifierFlags:[anEvent modifierFlags]
+                                                                                  timestamp:[anEvent timestamp]
+                                                                               windowNumber:_windowNumber
+                                                                                    context:nil
+                                                                                eventNumber:-1
+                                                                               trackingArea:aTrackingArea]];
+                }
+                
+                if (trackingOptions & CPTrackingCursorUpdate)
+                    [cursorUpdateStack addObject:aTrackingArea];
+            }
+        }
+    }
+    
+    // Search now for exited views (were in _mouseEnteredStack but no more in mouseEnteredStack)
+    
+    for (var i = 0; i < _mouseEnteredStack.length; i++)
+    {
+        var aTrackingArea   = _mouseEnteredStack[i],
+            trackingOptions = [aTrackingArea options];
+        
+        if (![mouseEnteredStack containsObject:aTrackingArea])
+        {
+            // Mouse is no more in this area so it's a mouseExited
+            
+            if ((trackingOptions & CPTrackingMouseEnteredAndExited) && (!dragging || (dragging && (trackingOptions & CPTrackingEnabledDuringMouseDrag))))
+                [[aTrackingArea owner] mouseExited:[CPEvent enterExitEventWithType:CPMouseExited
+                                                                          location:point
+                                                                     modifierFlags:[anEvent modifierFlags]
+                                                                         timestamp:[anEvent timestamp]
+                                                                      windowNumber:_windowNumber
+                                                                           context:nil
+                                                                       eventNumber:-1
+                                                                      trackingArea:aTrackingArea]];
+            
+            // If this is the active cursor area, we reset _cursorUpdateStack so a new active area will be computed
+            
+            if (aTrackingArea === _activeCursorTrackingArea)
+            {
+                _cursorUpdateStack = [];
+                _activeCursorTrackingArea = nil;
+            }
+        }
+    }
+
+    _mouseEnteredStack = mouseEnteredStack;
+
+    if (dragging)
+        
+        // While dragging, don't update the cursor
+        return;
+    
+    // Cursor update
+    
+    // We must send the cursorUpdate event to the frontmost view (the nearest from the mouse pointer)
+    // If no frontmost view is found, we set standard cursor
+    
+    var nbCursorUpdates          = [cursorUpdateStack count],
```

Comment:
Well, I was at first doing this via hitTest but this will find a view, not a tracking area. You may have conditions where a view without a tracking area is first, a tracking area second and then hitTest will return the first one. Moreover, as every window rects are already calculated, calling hitTest (which does all transformations) would be a large overhead. This is why I've written the present code (dealing only with tracking areas). Maybe a little "tricky" but able to deal with all cases I've tested.